src/System.Memory/tests/ReadOnlyMemory/Strings.cs

```diff
@@ -0,0 +1,175 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+using System.Buffers;
+using System.Collections.Generic;
+
+namespace System.MemoryTests
+{
+    public static partial class ReadOnlyMemoryTests
+    {
+        public static IEnumerable<object[]> StringInputs()
+        {
+            yield return new object[] { "" };
+            yield return new object[] { "a" };
+            yield return new object[] { "a\0bcdefghijklmnopqrstuvwxyz" };
+        }
+
+        [Theory]
+        [MemberData(nameof(StringInputs))]
+        public static void AsReadOnlyMemory_ToArray_Roundtrips(string input)
+        {
+            ReadOnlyMemory<char> m = input.AsReadOnlyMemory();
+            Assert.Equal(input, new string(m.ToArray()));
+        }
+
+        [Theory]
+        [MemberData(nameof(StringInputs))]
+        public static void AsReadOnlyMemory_Span_Roundtrips(string input)
+        {
+            ReadOnlyMemory<char> m = input.AsReadOnlyMemory();
+            ReadOnlySpan<char> s = m.Span;
+            Assert.Equal(input, new string(s.ToArray()));
+        }
+
+        [Theory]
+        [InlineData("", 0, 0)]
+        [InlineData("0123456789", 0, 0)]
+        [InlineData("0123456789", 10, 0)]
+        [InlineData("0123456789", 0, 10)]
+        [InlineData("0123456789", 1, 9)]
+        [InlineData("0123456789", 2, 8)]
+        [InlineData("0123456789", 9, 1)]
+        [InlineData("0123456789", 1, 8)]
+        [InlineData("0123456789", 5, 3)]
+        public static void AsReadOnlyMemory_Slice_MatchesSubstring(string input, int offset, int count)
+        {
+            ReadOnlyMemory<char> m = input.AsReadOnlyMemory();
+            Assert.Equal(input.Substring(offset, count), new string(m.Slice(offset, count).ToArray()));
+            Assert.Equal(input.Substring(offset, count), new string(m.Slice(offset, count).Span.ToArray()));
+            if (offset + count == input.Length)
+            {
+                Assert.Equal(input.Substring(offset), new string(m.Slice(offset).ToArray()));
+            }
+        }
+
+        [Fact]
+        public static void AsReadOnlyMemory_NullString_Throws()
+        {
+            AssertExtensions.Throws<ArgumentNullException>("text", () => ((string)null).AsReadOnlyMemory());
+        }
+
+        [Fact]
+        public static void AsReadOnlyMemory_TryGetString_Roundtrips()
+        {
+            string input = "0123456789";
+            ReadOnlyMemory<char> m = input.AsReadOnlyMemory();
+            Assert.False(m.IsEmpty);
+
+            string text;
+            int start;
+            int length;
+
+            Assert.True(m.TryGetString(out text, out start, out length));
```

Comment:
use inline declaration of the out parameters?

Reply:
I didn't want this block to be different in that regard from the subsequent blocks that also all write to these, so I separated them out.  I also find the out var declaration difficult to read in anything other than tiny methods, because when I want to see where something is declared, I have to essentially read all of the code in the method rather than just looking for the few places they could have been declared.